### PR TITLE
PointerLockControls: Refactoring

### PR DESCRIPTION
**Pull request by Mugen87**
Meanwhile, Pointer Lock API is [supported by all major browsers](https://caniuse.com/#feat=pointerlock). This is a good opportunity in order to simplify `misc_controls_pointerlock`  and refactor the API of `PointerLockControls`.

https://rawgit.com/Mugen87/three.js/dev2/examples/misc_controls_pointerlock.html

### CODE REVIEW

File: examples/misc_controls_pointerlock.html [hunk 1]
```diff
@@ -173,6 +108,30 @@
...
+				}, false );
+
+				controls.onLock = function() {
+
+					instructions.style.display = 'none';
+					blocker.style.display = 'none';
+
+				};
+
+				controls.onExit = function() {
```
Comment (mrdoob) [inline examples/misc_controls_pointerlock.html]: onUnlock maybe?

Comment (Mugen87) [reply examples/misc_controls_pointerlock.html]: :+1:

**Comment by mrdoob** [comment]
Actually, another change...

I think it would be better to make it event based instead:

```js
controls.addEventListener( 'lock', function ( event ) {

 	instructions.style.display = 'none';
	blocker.style.display = 'none';

} );

controls.addEventListener( 'unlock', function ( event ) {

	instructions.style.display = '';
	blocker.style.display = 'block';

} );
```

**Comment by Mugen87** [comment]
Yeah, that's more flexible :+1:

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
@Mugen87 FYI 6a1364282fbf44410eedcfb720da7d0e07f3fef7

**Pull request merged**